### ISSUE TITLE
Can not use keys with common part(without dot notation).

### ISSUE THREAD

**Issue by janowsiany** [meteor/meteor#9167]
When running on `METEOR@1.6-rc.1` i encounter strange problem when executing following operation:
```js
import {Mongo} from 'meteor/mongo';

const Todos = new Mongo.Collection('todos');

Todos.update({key: 123, keyName: '321'}, {$set: {name: 'Todo'}}, {upsert: true});
```
then the error is:
```
Error: cannot infer query fields to set, both paths 'key' and 'keyName' are matched
    at Object.keys.forEach.existingKey (packages/minimongo/common.js:782:13)
    at Array.forEach (<anonymous>)
    at insertIntoDocument (packages/minimongo/common.js:777:25)
    at populateDocumentWithKeyValue (packages/minimongo/common.js:1189:5)
    at Object.keys.forEach.key (packages/minimongo/common.js:1243:9)
    at Array.forEach (<anonymous>)
    at populateDocumentWithQueryFields (packages/minimongo/common.js:1228:24)
    at Function.LocalCollection._createUpsertDocument (packages/minimongo/local_collection.js:921:28)
    at MongoConnection._update (packages/mongo/mongo_driver.js:534:38)
    at MongoConnection.<anonymous> (packages/meteor.js:208:21)
```

I beleive that this [code](https://github.com/meteor/meteor/blob/devel/packages/minimongo/common.js#L780-L781), introduced in #8815, should only affect keys in dot notation. I am eager to fix this if confirmed. Just wondering if there are other places which depend on this behaviour?
@hwillson @benjamn @abernix

**Comment by janowsiany** [comment]
call @sebakerckhof

**Comment by sebakerckhof** [comment]
yep, your analysis is correct.

**Comment by benjamn** [comment]
Is this fixed in `1.6-rc.8`?

**Comment by klaussner** [comment]
> Is this fixed in `1.6-rc.8`?

Yes. :tada:
